Unable to mount configuration.yml

I tried to install authelia with docker-compose but on startup authelia isn't able to find the configuration.yml. Especially it is impossible to mount the "/var/lib/authelia" volumen and on build the container automatically mounts /etc/authelia (where the configuration files are in. But even on changing the PUID or PGID to root or change the configuration-files in the (unexpected) volumen, authelia still goes in panic mode and logs missing configuration.

I even copied your docker-compose.yml and still the same error. So either I'm doing something stupidly wrong (probably) or there is a bug (unlikely). 

```
time="2020-05-26T17:25:41+02:00" level=error msg="Provide a JWT secret using \"jwt_secret\" key"
time="2020-05-26T17:25:41+02:00" level=error msg="Please provide `ldap` or `file` object in `authentication_backend`"
time="2020-05-26T17:25:41+02:00" level=error msg="Set domain of the session object"
time="2020-05-26T17:25:41+02:00" level=error msg="A storage configuration must be provided. It could be 'local', 'mysql' or 'postgres'"
time="2020-05-26T17:25:41+02:00" level=error msg="A notifier configuration must be provided"
panic: Some errors have been reported

goroutine 1 [running]:
main.startServer()
	github.com/authelia/authelia/cmd/authelia/main.go:41 +0xc80
main.main.func1(0xc00009c000, 0xc000232120, 0x0, 0x2)
	github.com/authelia/authelia/cmd/authelia/main.go:126 +0x20
github.com/spf13/cobra.(*Command).execute(0xc00009c000, 0xc000020190, 0x2, 0x2, 0xc00009c000, 0xc000020190)
	github.com/spf13/cobra@v0.0.7/command.go:842 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc00009c000, 0xc00013df58, 0x4, 0x4)
	github.com/spf13/cobra@v0.0.7/command.go:943 +0x317
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v0.0.7/command.go:883
main.main()
	github.com/authelia/authelia/cmd/authelia/main.go:143 +0x166
```

This:
```
/etc/authelia # authelia --config /etc/authelia/configuration.yml
ERRO[0000] Provide a JWT secret using "jwt_secret" key  
ERRO[0000] Please provide `ldap` or `file` object in `authentication_backend` 
ERRO[0000] Set domain of the session object             
ERRO[0000] A storage configuration must be provided. It could be 'local', 'mysql' or 'postgres' 
ERRO[0000] A notifier configuration must be provided    
panic: Some errors have been reported

goroutine 1 [running]:
main.startServer()
        github.com/authelia/authelia/cmd/authelia/main.go:41 +0xc80
main.main.func1(0xc000376580, 0xc000272060, 0x0, 0x2)
        github.com/authelia/authelia/cmd/authelia/main.go:126 +0x20
github.com/spf13/cobra.(*Command).execute(0xc000376580, 0xc000020190, 0x2, 0x2, 0xc000376580, 0xc000020190)
        github.com/spf13/cobra@v0.0.7/command.go:842 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc000376580, 0xc0000b5f58, 0x4, 0x4)
        github.com/spf13/cobra@v0.0.7/command.go:943 +0x317
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v0.0.7/command.go:883
main.main()
        github.com/authelia/authelia/cmd/authelia/main.go:143 +0x166
/etc/authelia # 
```

--- a/cmd/authelia/coverage_test.go
+++ b/cmd/authelia/coverage_test.go
@@ -10,7 +10,7 @@ import (
 	"testing"
 )
 
-func TestAuthelia(t *testing.T) {
+func TestCoverage(t *testing.T) {
 	var (
 		args []string
 	)

--- a/cmd/authelia/main.go
+++ b/cmd/authelia/main.go
@@ -1,7 +1,6 @@
 package main
 
 import (
-	"errors"
 	"fmt"
 	"log"
 	"os"
@@ -27,10 +26,6 @@ var configPathFlag string
 
 //nolint:gocyclo // TODO: Consider refactoring/simplifying, time permitting
 func startServer() {
-	if configPathFlag == "" {
-		log.Fatal(errors.New("No config file path provided"))
-	}
-
 	config, errs := configuration.Read(configPathFlag)
 
 	if len(errs) > 0 {
@@ -38,7 +33,7 @@ func startServer() {
 			logging.Logger().Error(err)
 		}
 
-		panic(errors.New("Some errors have been reported"))
+		os.Exit(1)
 	}
 
 	if err := logging.InitializeLogger(config.LogFilePath); err != nil {

--- a/internal/configuration/reader.go
+++ b/internal/configuration/reader.go
@@ -1,10 +1,14 @@
 package configuration
 
 import (
+	"errors"
 	"fmt"
+	"io/ioutil"
+	"os"
 	"strings"
 
 	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
 
 	"github.com/authelia/authelia/internal/configuration/schema"
 	"github.com/authelia/authelia/internal/configuration/validator"
@@ -12,6 +16,27 @@ import (
 
 // Read a YAML configuration and create a Configuration object out of it.
 func Read(configPath string) (*schema.Configuration, []error) {
+	if configPath == "" {
+		return nil, []error{errors.New("No config file path provided")}
+	}
+
+	_, err := os.Stat(configPath)
+	if err != nil {
+		return nil, []error{fmt.Errorf("Unable to find config file: %v", configPath)}
+	}
+
+	file, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return nil, []error{fmt.Errorf("Failed to %v", err)}
+	}
+
+	var data interface{}
+
+	err = yaml.Unmarshal(file, &data)
+	if err != nil {
+		return nil, []error{fmt.Errorf("Error malformed %v", err)}
+	}
+
 	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 
 	viper.BindEnv("authelia.jwt_secret.file")                           //nolint:errcheck // TODO: Legacy code, consider refactoring time permitting.
@@ -25,11 +50,7 @@ func Read(configPath string) (*schema.Configuration, []error) {
 
 	viper.SetConfigFile(configPath)
 
-	if err := viper.ReadInConfig(); err != nil {
-		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
-			return nil, []error{fmt.Errorf("unable to find config file %s", configPath)}
-		}
-	}
+	_ = viper.ReadInConfig()
 
 	var configuration schema.Configuration
 

--- a/internal/configuration/reader_test.go
+++ b/internal/configuration/reader_test.go
@@ -57,6 +57,39 @@ func setupEnv(t *testing.T) string {
 	return dir
 }
 
+func TestShouldErrorNoConfigPath(t *testing.T) {
+	_, errors := Read("")
+
+	require.Len(t, errors, 1)
+
+	require.EqualError(t, errors[0], "No config file path provided")
+}
+
+func TestShouldErrorNoConfigFile(t *testing.T) {
+	_, errors := Read("./nonexistent.yml")
+
+	require.Len(t, errors, 1)
+
+	require.EqualError(t, errors[0], "Unable to find config file: ./nonexistent.yml")
+}
+
+func TestShouldErrorPermissionsConfigFile(t *testing.T) {
+	_ = ioutil.WriteFile("/tmp/authelia/permissions.yml", []byte{}, 0000) // nolint:gosec
+	_, errors := Read("/tmp/authelia/permissions.yml")
+
+	require.Len(t, errors, 1)
+
+	require.EqualError(t, errors[0], "Failed to open /tmp/authelia/permissions.yml: permission denied")
+}
+
+func TestShouldErrorParseBadConfigFile(t *testing.T) {
+	_, errors := Read("./test_resources/config_bad_quoting.yml")
+
+	require.Len(t, errors, 1)
+
+	require.EqualError(t, errors[0], "Error malformed yaml: line 23: did not find expected alphabetic or numeric character")
+}
+
 func TestShouldParseConfigFile(t *testing.T) {
 	dir := setupEnv(t)
 

--- a/internal/configuration/test_resources/config_bad_quoting.yml
+++ b/internal/configuration/test_resources/config_bad_quoting.yml
@@ -1,0 +1,52 @@
+host: 0.0.0.0
+port: 9091
+log_level: debug
+
+jwt_secret: RUtG9TnbXrOl1XLLmDgySw1DGgx9QcrtepIf1uDDBlBVKFZxkVBruYKBi32PvaU
+
+default_redirection_url: example.com
+
+totp:
+  issuer: example.com
+  period: 30
+  skew: 1
+
+authentication_backend:
+  file:
+    path: /etc/authelia/users_database.yml
+
+access_control:
+  default_policy: deny
+  rules:
+    - domain: example.com
+      policy: bypass
+    - domain: *.example.com
+      policy: one_factor
+
+session:
+  name: authelia_session
+  secret: TVPMIcDFbBwhnW3kLJzKhdjeHhtqisr7m28FgRY8oLh2A4lwuV2jV2ZGdGbh4aa
+  expiration: 3600
+  inactivity: 300
+  domain: example.com
+
+regulation:
+  max_retries: 3
+  find_time: 120
+  ban_time: 300
+
+storage:
+  mysql:
+    host: example.com
+    port: 3306
+    database: authelia
+    username: authelia
+    password: example.com
+
+notifier:
+  smtp:
+    username: example.com
+    password: example.com
+    host: smtp.gmail.com
+    port: 465
+    sender: example.com

